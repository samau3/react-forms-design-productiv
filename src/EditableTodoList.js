import React from "react";
import EditableTodo from "./EditableTodo";

/** Show list of editable todos.
 *
 * Props:
 * - todos: array of [ todo, ... ]
 * - update(): fn to call to update a todo
 * - remove(): fn to call to remove a todo
 *
 * TodoApp -> EditableTodoList -> [ EditableTodo, ... ]
 */

function EditableTodoList({ todos, update, remove }) {

  function updateItem() {
    update(todo.id)
  }

  function removeItem() {
    remove(todo.id)
  }

  return (
    <div>
      {/*   FIXME  */}
      {todos.map(todo =>
        <div>
          <EditableTodo todo={todo} />
          <button onClick={updateItem}>Edit</button>
          <button onClick={removeItem}>Remove</button>
        </div>
      )}
    </div>
  );
}

export default EditableTodoList;
