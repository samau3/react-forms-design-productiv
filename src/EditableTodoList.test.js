import React from "react";
import { render } from "@testing-library/react";

import TEST_DATA from "./_testCommon";
import EditableTodoList from "./EditableTodoList";

it("renders without crashing", function () {
    render(<EditableTodoList todos={TEST_DATA} />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<EditableTodoList todos={TEST_DATA} />);
    expect(asFragment()).toMatchSnapshot();
});