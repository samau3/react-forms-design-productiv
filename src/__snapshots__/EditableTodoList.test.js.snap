// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="EditableTodo"
        id="1"
      >
        <div
          class="mb-3"
        >
          <div
            class="float-right text-sm-right"
          >
            <div
              class="Todo"
            >
              <div>
                <b>
                  Code!
                </b>
                 
                <small>
                  (priority: 2)
                </small>
              </div>
              <div>
                <small>
                  Write some code
                </small>
              </div>
            </div>
            <button
              class="EditableTodo-toggle btn-link btn btn-sm"
            >
              Edit
            </button>
            <button
              class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            >
              Del
            </button>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="EditableTodo"
        id="2"
      >
        <div
          class="mb-3"
        >
          <div
            class="float-right text-sm-right"
          >
            <div
              class="Todo"
            >
              <div>
                <b>
                  Make dinner
                </b>
                 
                <small>
                  (priority: 1)
                </small>
              </div>
              <div>
                <small>
                  Cook something healthy
                </small>
              </div>
            </div>
            <button
              class="EditableTodo-toggle btn-link btn btn-sm"
            >
              Edit
            </button>
            <button
              class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            >
              Del
            </button>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="EditableTodo"
        id="3"
      >
        <div
          class="mb-3"
        >
          <div
            class="float-right text-sm-right"
          >
            <div
              class="Todo"
            >
              <div>
                <b>
                  Go to bed
                </b>
                 
                <small>
                  (priority: 3)
                </small>
              </div>
              <div>
                <small>
                  In bed by 11:15
                </small>
              </div>
            </div>
            <button
              class="EditableTodo-toggle btn-link btn btn-sm"
            >
              Edit
            </button>
            <button
              class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
            >
              Del
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
